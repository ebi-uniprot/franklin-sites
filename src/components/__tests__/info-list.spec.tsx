import { render } from '@testing-library/react';

import InfoList from '../info-list';

describe('InfoList component', () => {
  const infoData = [
    {
      title: 'Item 1',
      content: <div>Some content</div>,
    },
    {
      title: 'Another item',
      content: <div>Some more content</div>,
    },
  ];
  it('should render', () => {
    const { asFragment } = render(<InfoList infoData={infoData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render nothing when empty list', () => {
    const { container } = render(<InfoList infoData={[]} />);
    expect(container).toBeEmptyDOMElement();
  });

  it('should render nothing when no content', () => {
    const { container } = render(
      <InfoList
        infoData={[
          { title: 'Item 1', content: null },
          { title: 'Item 2', content: undefined },
          { title: 'Item 3', content: '' },
        ]}
      />
    );
    expect(container).toBeEmptyDOMElement();
  });

  it('should render compact', () => {
    const { asFragment } = render(<InfoList infoData={infoData} isCompact />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without titles', () => {
    const { asFragment } = render(<InfoList infoData={infoData} noTitles />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with titles in 2 columns', () => {
    const { asFragment } = render(<InfoList infoData={infoData} columns />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not render items that are undefined', () => {
    const { asFragment } = render(
      <InfoList
        infoData={[
          { title: 'Item 1', content: 'content' },
          { title: 'Item 2', content: undefined },
        ]}
        columns
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render title that is a React Node', () => {
    const { asFragment } = render(
      <InfoList
        infoData={[
          { title: <div>Item 1</div>, content: <div>Some content</div> },
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
