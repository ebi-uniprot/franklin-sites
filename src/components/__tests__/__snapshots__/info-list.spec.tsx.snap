// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoList component should not render items that are undefined 1`] = `
<DocumentFragment>
  <ul
    class="info-list info-list--columns"
  >
    <li>
      <div
        class="decorated-list-item"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Item 1
        </div>
        <div
          class="decorated-list-item__content"
        >
          content
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`InfoList component should render 1`] = `
<DocumentFragment>
  <ul
    class="info-list"
  >
    <li>
      <div
        class="decorated-list-item"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Item 1
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some content
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        class="decorated-list-item"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Another item
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some more content
          </div>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`InfoList component should render compact 1`] = `
<DocumentFragment>
  <ul
    class="info-list"
  >
    <li>
      <div
        class="decorated-list-item decorated-list-item--compact"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Item 1
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some content
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        class="decorated-list-item decorated-list-item--compact"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Another item
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some more content
          </div>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`InfoList component should render title that is a React Node 1`] = `
<DocumentFragment>
  <ul
    class="info-list"
  >
    <li>
      <div
        class="decorated-list-item"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          <div>
            Item 1
          </div>
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some content
          </div>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`InfoList component should render with titles in 2 columns 1`] = `
<DocumentFragment>
  <ul
    class="info-list info-list--columns"
  >
    <li>
      <div
        class="decorated-list-item"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Item 1
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some content
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        class="decorated-list-item"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Another item
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some more content
          </div>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`InfoList component should render without titles 1`] = `
<DocumentFragment>
  <ul
    class="info-list"
  >
    <li>
      <div
        class="decorated-list-item decorated-list-item--no-title"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Item 1
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some content
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        class="decorated-list-item decorated-list-item--no-title"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Another item
        </div>
        <div
          class="decorated-list-item__content"
        >
          <div>
            Some more content
          </div>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
